Source/SwiftLintBuiltInRules/Rules/Idiomatic/OneDelarationPerFileRule.swift

```diff
@@ -0,0 +1,118 @@
+import Foundation
+import SwiftSyntax
+
+@SwiftSyntaxRule
+struct OneDelarationPerFileRule: Rule {
+    var configuration = SeverityConfiguration<Self>(.error)
+
+    static let description = RuleDescription(
+        identifier: "one_declaration_per_file",
+        name: "One Declaration Per File",
+        description: "One declaration per file is allowed, extensions are an exception",
+        kind: .idiomatic,
+        nonTriggeringExamples: [
+            Example("""
+                    class Car {
+                        var make: String
+                        var model: String
+                        init(make: String, model: String) {
```

Comment:
The contents of the declarations in all the examples aren't required to document and test the rule. Let's have very short examples like

```swift
class C {}
extension E {}
```